bpf/node_config.h

```diff
@@ -176,6 +176,12 @@ DEFINE_IPV6(HOST_IP, 0xbe, 0xef, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0xa, 0x
 # define LB_SELECTION		LB_SELECTION_RANDOM
 #endif
 
+#ifdef ENABLE_VTEP
+#define VTEP_ENDPOINT (__u32[]){0xec48a90a, 0xee48a90a, }
+#define VTEP_MAC (__u64[]){0x562e984c3682, 0x552e984c3682, }
+#define VTEP_NUMS 4
```

Comment:
^^ This shouldn't be a problem for the real code since the commandline flag parsing and validation code looks good and looks like it will prevent this kind of situation. But I don't understand how we could compile the above code and load it into the kernel at all in this state, because presumably the loops could iterate up to index 3 and then attempt to pull random values out of either the macro above or maybe random stack values outside the bounds of the arrays declared above...?

Reply:
> The `VTEP_ENDPOINT` and `VTEP_MAC` variables above seem to have two values, but `VTEP_NUMS` is `4`. Is this intentional?

I was thinking about limiting the number of VTEP `<=4`, but `<=2` would work too, we usually have two BIG-IPs as active/standby pair.
> 
> This code is only used when running the bpf-verifier tests, are those tests currently covering this case?

I will change the `VTEP_NUMS` to `2` to match the number of `VTEP_ENDPOINT`, sounds good?

Reply:
That's fine, but my concern is more, "Is this being tested?" . The discrepancy between these different #defines suggests to me that this code is not being tested (or maybe I just don't understand what the testing is doing).

The reason that I advocate for defining these macros here is that ideally the verifier tests from ginkgo `test/k8sT/Verifier.go` will attempt to compile and load the most complex possible eBPF programs and validate that the kernel will happily load these programs. Then these tests will detect complexity issues before we hand this code to users, and allow the Cilium development community to come up with solutions to solve these complexity issues without the pressure of having bugs out there actively affecting users.

Reply:
^^ To extend, so intend for this feature to support up to four VTEPs. Worst possible case, if we set this to four and load the eBPF programs, that may introduce X additional instructions into the datapath, which the kernel must verify to be safe. So we should set this file to the most complicated settings that may end up being used at runtime, and then make sure that the testing will attempt to build & load the programs with these settings, so that the tests tell us about problems before users do.

Reply:
@joestringer  right, that makes sense. I am not clear though how to test it, I did run `KERNEL=54 make -C bpf/` after adding the macro in `bpf/node_config.h`, I think I also tested adding 4 VTEPs in my local lab with cilium agent and no problem, but I think you mean some automated test with `test/k8sT/Verifier.go` that I am not familiar with, could you elaborate on how to achieve that?

Reply:
or if I understand it correct, you mean I should add 4 `VTEP_ENDPOINT` and 4 `VTELP_MAC` in `bpf/node_config.h` and the `test/k8sT/Verifier.go` will be automatically invoked after pulling my changes to do the compile and load test, if that is the case, yes, I will add 4 VTEPs.